From f470814b10d77535822bd70df3153a918a53e0c3 Mon Sep 17 00:00:00 2001
From: Shawn Lin <shawn.lin@rock-chips.com>
Date: Fri, 27 Nov 2020 10:23:47 +0800
Subject: [PATCH] PCI: rockchip: dw: Remove link speed setting

It's should be called by dwc core driver by walking up
the capability registers.

Change-Id: I27ff106dc7688734339c627e32360841d58d68a5
Signed-off-by: Shawn Lin <shawn.lin@rock-chips.com>
---
 drivers/pci/controller/dwc/pcie-dw-rockchip.c | 21 -------------------
 1 file changed, 21 deletions(-)

diff --git a/drivers/pci/controller/dwc/pcie-dw-rockchip.c b/drivers/pci/controller/dwc/pcie-dw-rockchip.c
index 6831897eaeaa..6454ed3837b9 100644
--- a/drivers/pci/controller/dwc/pcie-dw-rockchip.c
+++ b/drivers/pci/controller/dwc/pcie-dw-rockchip.c
@@ -122,7 +122,6 @@ struct rk_pcie {
 	bool				in_suspend;
 	bool				is_rk1808;
 	bool				bifurcation;
-	int				link_gen;
 	struct regulator		*vpcie3v3;
 };
 
@@ -360,20 +359,6 @@ static inline void rk_pcie_enable_ltssm(struct rk_pcie *rk_pcie)
 	rk_pcie_writel_apb(rk_pcie, 0x0, 0xC000C);
 }
 
-static inline void rk_pcie_set_gens(struct rk_pcie *rk_pcie)
-{
-	int gen_encode[] = {0x1, 0x2, 0x4};
-
-	if (rk_pcie->link_gen <= 0 || rk_pcie->link_gen > 3)
-		rk_pcie->link_gen = gen_encode[1]; /* Default to Gen2 */
-	else
-		rk_pcie->link_gen = gen_encode[rk_pcie->link_gen - 1];
-
-	dw_pcie_writel_dbi(rk_pcie->pci,
-			   PCIE_CAP_LINK_CONTROL2_LINK_STATUS,
-			   rk_pcie->link_gen);
-}
-
 static int rk_pcie_link_up(struct dw_pcie *pci)
 {
 	struct rk_pcie *rk_pcie = to_rk_pcie(pci);
@@ -1167,10 +1152,6 @@ static int rk_pcie_probe(struct platform_device *pdev)
 	/* Set PCIe mode */
 	rk_pcie_set_mode(rk_pcie);
 
-	/* Set PCIe gen */
-	rk_pcie->link_gen = of_pci_get_max_link_speed(np);
-	rk_pcie_set_gens(rk_pcie);
-
 	switch (rk_pcie->mode) {
 	case RK_PCIE_RC_TYPE:
 		ret = rk_add_pcie_port(rk_pcie);
@@ -1270,8 +1251,6 @@ static int __maybe_unused rockchip_dw_pcie_resume(struct device *dev)
 
 	/* Set PCIe mode */
 	rk_pcie_set_mode(rk_pcie);
-	/* Set PCIe gen */
-	rk_pcie_set_gens(rk_pcie);
 
 	ret = rk_pcie_establish_link(rk_pcie->pci);
 	if (ret) {
-- 
2.35.3

